src/Common/CommonTable.tsx

```diff
@@ -0,0 +1,162 @@
+import { FaEdit, FaTrash, FaDownload, FaAd, FaPlusCircle } from 'react-icons/fa'; // Import Font Awesome icons
+import '../Styles/CommonTable.css';
+import DeleteConfirmation from './DeleteConfirmation';
+import '../Styles/Room.css';
+import { format } from 'date-fns';
+
+
+interface CommonTableProps {
+  columns: any[];
+  data: any[];
+  handleEdit?: (rowData: any) => void; // Make handleEdit optional
+  handleDelete?: (rowData: any) => void; // Make handleDelete optional
+  downloadInvoice?: (rowData: any) => void; // Make download optional
+  showTretment?: (rowData: any) => void; 
+}
+
+const CommonTable: React.FC<CommonTableProps> = ({ columns, data, handleEdit, handleDelete, downloadInvoice,showTretment }) => {
+  const [showModal, setShowModal] = useState(false);
+  const [selectedRow, setSelectedRow] = useState<any>(null); // Specify type for selectedRow
+
+  const downloadInvoiceIn = (rowData: any) => {
+    setSelectedRow(rowData);
+    if (downloadInvoice) {
+      downloadInvoice(rowData);
+    }
+  };
+
+  const handleDeleteIn = () => {
+    setShowModal(false);
+    if (handleDelete) {
+      handleDelete(selectedRow);
+    }
+  };
+
+  const openDialogForDelete = (rowData: any) => {
+    setShowModal(true);
+    setSelectedRow(rowData);
+  };
+
+  const columnsWithActions = useMemo(() => {
+    const actionsColumn = {
+      Header: 'Actions',
+      accessor: 'actions',
+      Cell: ({ row }: any) => (
+        <div className='text-center'>
+          {handleEdit && (
+            <FaEdit onClick={() => handleEdit(row.original)} className='action-button cp m-1' />
+          )}
+          {handleDelete && (
+            <FaTrash onClick={() => openDialogForDelete(row.original)} className='action-button cp m-1' />
+          )}
+          {downloadInvoice && (
+            <FaDownload onClick={() => downloadInvoiceIn(row.original)} className='action-button cp m-1' />
+          )}
+        </div>
+      ),
+    };
+    const actionsColumnWithTreatment = {
+      Header: 'Treatment',
+      accessor: 'treatment',
+      Cell: ({ row }: any) => (
+        <div className='text-center'>
+            {showTretment &&<FaPlusCircle onClick={() => showTretment(row.original)} className='action-button cp m-1'/>}
+        </div>
+      ),
+    };
+
+    if(showTretment){
+      return [...columns, actionsColumn,actionsColumnWithTreatment ]
+    }
+    return handleEdit || handleDelete ? [...columns, actionsColumn] : columns;
+  }, [columns, handleEdit, handleDelete]);
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    prepareRow,
+    page,
+    nextPage,
+    previousPage,
+    canNextPage,
+    canPreviousPage,
+    pageOptions,
+    state: { pageIndex },
+  } = useTable(
+    {
+      columns: columnsWithActions,
+      data,
+      initialState: { pageIndex: 0 } as any,
+    },
+    usePagination
+  ) as any;
+
+
+  return (
+    <div className="common-table">
```

Comment:
move this JSX into separate component

Reply:
This is already in seperate common component